docs/client-concepts/high-level/indexing/indexing.asciidoc

```diff
@@ -0,0 +1,186 @@
+
+==== Single Documents
+
+A single document can be indexed at a time, either synchronously or asynchronously
+
+[source,csharp]
+----
+var person = new Person
+{
+    Id = 1,
+    FirstName = "Martijn",
+    LastName = "Laarman"
+};
+
+var indexResponse = client.IndexDocument(person); <1>
+
+var asyncIndexResponse = await client.IndexDocumentAsync(person); <2>
+----
+<1> synchronous method that returns an `IIndexResponse`
+
+<2> asynchronous method that returns a `Task<IIndexResponse>` that can be awaited
+
+==== Multiple Documents with IndexMany
+
+Multiple documents can be indexed using the `IndexMany` and `IndexManyAsync` methods, again either synchronously or asynchronously.
+These methods are specific to the NEST client and wrap calls to the `_bulk` endpoint, providing a convenient shortcut to indexing
+multiple documents.
+
+[source,csharp]
+----
+var people = new []
+{
+    new Person
+    {
+        Id = 1,
+        FirstName = "Martijn",
+        LastName = "Laarman"
+    },
+    new Person
+    {
+        Id = 2,
+        FirstName = "Stuart",
+        LastName = "Cam"
+    },
+    new Person
+    {
+        Id = 3,
+        FirstName = "Russell",
+        LastName = "Cam"
+    }
+};
+
+var indexManyResponse = client.IndexMany(people); <1>
+
+if (indexManyResponse.Errors) <2>
+{
+    <3>
+    foreach (var itemWithError in indexManyResponse.ItemsWithErrors)
+    {
+        Console.WriteLine("Failed to index document {0}: {1}", itemWithError.Id, itemWithError.Error);
+    }
+}
+
+// Alternatively, documents can be indexed asynchronously
+var indexManyAsyncResponse = await client.IndexManyAsync(people); <4>
+----
+<1> synchronous method that returns an `IBulkResponse`
+
+<2> the response can be inspected for its success
+
+<3> If there are errors, they can be enumerated and inspected
+
+<4> asynchronous method that returns a `Task<IBulkResponse>` that can be awaited
+
+==== Multiple Documents with Bulk
+
+If you require finer grained control over bulk indexing you can use the `Bulk` and `BulkAsync` methods and use the descriptors to
+customise the bulk calls.
+
+[source,csharp]
+----
+var bulkIndexResponse = client.Bulk(b => b
+    .Index("people")
+    .IndexMany(people)); <1>
+
+// Alternatively, documents can be indexed asynchronously similar to `IndexManyAsync`
+var asyncBulkIndexResponse = await client.BulkAsync(b => b
+    .Index("people")
+    .IndexMany(people)); <4>
+----
+<1> synchronous method that returns an `IBulkResponse`, the same as `IndexMany` and can be inspected in the same way for errors
+
+<4> asynchronous method that returns a `Task<IBulkResponse>` that can be awaited
+
+==== Multiple Documents with BulkAllObservable helper
+
+Multiple documents can be indexed using the `BulkAllObservable` helper. This helper exposes retry and backoff functionality
+to automatically retry in the event of a failure. This allows you to focus on the overall objective of indexing,
```

Comment:
should also mention that it

1. sends multiple concurrent bulk requests each with a number of documents specified by size
2. will lazily enumerate an `IEnumerable<T>` collection of documents, allowing to index a large corpus of documents easily.